alarm_active_table.cpp

```diff
@@ -0,0 +1,423 @@
+    {
+      // We do not support multiple contexts so this object
+      // will be set to the empty string
+      static std::string empty_string  = "";
+      snmp_set_var_typed_value(var, ASN_OCTET_STR,
+                               (u_char*) &empty_string,
+                               empty_string.length());
+    }
+    break;
+    
+    case COLUMN_ALARMACTIVEVARIABLES:
+    {
+      // The number of variables in alarmActiveVariableTable for
+      // this alarm. See as we have no such table this will be 0.
+      static unsigned long variables = 0;
+      snmp_set_var_typed_value(var, ASN_UNSIGNED,
+                               (u_char*) &variables,
+                               sizeof(variables));
+    }
+    break;
+
+    case COLUMN_ALARMACTIVENOTIFICATIONID:
+    {
+      if (context->_alarm_table_def->severity() == AlarmDef::CLEARED)
+      {
+        snmp_set_var_typed_value(var, ASN_OBJECT_ID,
+                                 (u_char*) alarm_clear_state_oid,
+                                 sizeof(alarm_clear_state_oid));
+      }
+      else
+      {
+        snmp_set_var_typed_value(var, ASN_OBJECT_ID,
+                                 (u_char*) alarm_active_state_oid,
+                                 sizeof(alarm_active_state_oid));
+      }
+    }
+    break;
+    
+    case COLUMN_ALARMACTIVERESOURCEID:
+    {
+      // None of our alarms have resources so this will always
+      // be the 0.0 OID
+      
+      snmp_set_var_typed_value(var, ASN_OBJECT_ID,
+                               (u_char*) zero_dot_zero_oid,
+                               sizeof(zero_dot_zero_oid));
+    }
+    break;
+    
+    case COLUMN_ALARMACTIVEDESCRIPTION:
+    {
+      snmp_set_var_typed_value(var, ASN_OCTET_STR,
+                               (u_char*) context->_alarm_table_def->description().c_str(),
+                               context->_alarm_table_def->description().length());
+    }
+    break;
+
+    case COLUMN_ALARMACTIVELOGPOINTER:
+    {
+      // We don't keep log entries so this will always be
+      // the 0.0 OID
+      snmp_set_var_typed_value(var, ASN_OBJECT_ID,
+                               (u_char*) zero_dot_zero_oid,
+                               sizeof(zero_dot_zero_oid));
+    }
+    break;
+
+    case COLUMN_ALARMACTIVEMODELPOINTER:
+    {
+      oid* model_pointer;
+      // Convert the index string OID
+      int model_pointer_len = alarmModelTable_oid_len + entry_column_oid_len + 3;
+      // Append the index array on to the Alarm Model Table OID, inserting
+      // the OID ".1.3." for the entry in the first non index column
+      model_pointer = new oid[model_pointer_len];
+      copy(alarmModelTable_oid, alarmModelTable_oid + alarmModelTable_oid_len, model_pointer);
+
+      copy(entry_column_oid, entry_column_oid + entry_column_oid_len, model_pointer + alarmModelTable_oid_len);
+      
+      model_pointer[alarmModelTable_oid_len + entry_column_oid_len] = 0;
```

Comment:
Would it be simpler just to add the `.1.3` for the entry columns like this? (I don't feel strongly.)